internal/expr/prepare.go

```diff
@@ -113,17 +124,27 @@ func prepareOutput(ti typeNameToInfo, p *outputPart) ([]fullName, []typeMember,
 		}
 		return info, nil
 	}
-
-	addColumns := func(info typeInfo, tag string, column fullName) error {
+	addColumns := func(info typeInfo, member string, column fullName) error {
 		var tm typeMember
 		switch info := info.(type) {
```

Comment:
I'm not a big fan of switch ...(type), vs using interfaces that have defined interactions.
eg:
```
type TypeMember interface {
  TypeMember(member string) (Res, error)
}

func (simpleType) TypeMember(member string) {
  if member != "" {
    return nil, fmt.Errorf(`cannot specify member of primitive type...')
  }
  return simpleType{...}, nil
}
```

That makes it far easier to extend your functionality in the future, because you can implement the interfaces, without having to change the hard coded switch statements everywhere.

Reply:
Very good point, I've refactored as suggested.